AmplifyPlugins/Internal/Sources/InternalAWSPinpoint/Context/PinpointContext.swift

```diff
@@ -110,7 +110,7 @@ class PinpointContext {
     private let storage: PinpointContextStorage
 
     init(with configuration: PinpointContextConfiguration,
-         endpointInformation: EndpointInformation = .current,
+         endpointInformationProvider: EndpointInformationProvider = DefaultEndpointInformationProvider(),
```

Comment:
is `EndpointInformation.current` the same as `DefaultEndpointInformationProvider` ?

Reply:
no, the provider will provide the current endpoint when it's needed, deferring the execution to get the current endpoint information to a latter point when we're already in an async context.  This avoids the need to make the PinpointContext constructor async